# DSS/DSS-hLZ Workflow and History

### Logging into Cabernet server:

```ssh klfurtado@cabernet.genomecenter.ucdavis.edu```

Will be prompted to enter password.

If personal account is not working, use:

```ssh eamaga@cabernet.genomecenter.ucdavis.edu```

Will be prompted to enter password.

### October 7th, 2016

Loaded files into the server from local hard drive.

From terminal on local computer:

```
165  scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS/ eamaga@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat
166  scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS-hlZ/ eamaga@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat
```

### October 21, 2016

Logged in using personal username and password:

``` ssh klfurtado@cabernet.genomecenter.ucdavis.edu```

Prompted to enter password.

Change into directory:

```cd share/tearlab/Maga/Kat```

*Permission denied for the following commands:*

Within server: ```mkdir```

From local terminal:
```
scp -r //Users/kathleenfurtado/Documents/maga.lab/dss/20150401_dss_mapping.file_contents.txt klfurtado@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat/DSS
```

### December 9th, 2016

Added sequence files to new genome center server directory.

```
ssh klfurtado@cabernet.genomecenter.ucdavis.edu
scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS/ klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat
scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS-hLZ/ klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat
```

### December 12, 2016

Created "Practice" directory for writing bash scripts, changed permissions, and executed command within same directory using ./

```
162  cd Kat
163  mkdir Practice
165  cd Practice
166  nano
167  nano awesome.sh
168  chmod u+x awesome.sh
169  ls -lah awesome.sh
173  ./awesome.sh
```

Unzipped fastq.gz files for DSS study. DSS-hLZ files were not unzipped (broken pipe?)

### December 13, 2016

Uploaded Mapping Files for DSS-hLZ. Unzipped DSS-hLZ data files.

### December 14, 2016

Wrote shell script to run flash2 and merge forward and reverse reads for DSS study.

```
212  mkdir merged.reads
213  cd merged.reads
214  nano flash2.dss.sh
215  nano flash2.dss.sh
216  ./flash2.dss.sh
217  chmod u+x flash2.dss.sh
218  ./flash2.dss.sh
219  cd /share/magalab/Kat
220  ls
221  cd DSS
222  ls
223  cd merged.reads
224  ls
225  less dss.feces.histogam
226  less dss.feces.histogram
```

Script: flash2.dss.sh

```
#!/bin/bash
##

echo "Setting up output and error directories"
#SBATCH -o /share/magalab/Kat/DSS/merged.reads/flash2.out
#SBATCG -e /share/magalab/Kat/DSS/merged.reads/flash2.err

echo "Loading Flash2 module"
module load flash2/c41a82e

#Average read length should be 250 bp, average fragment length should be between 290 and 300, and quality shou$

echo "Options -r 250, -f 290, and -s 29 are used to calculate max overlap"

time flash2 /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq -o$

echo "Merging should be complete"
```

Allowed to run until December 15, received error:

```
packet_write_wait: Connection to 128.120.143.100 port 22: Broken pipe
```

Several output files were still created.

### December 19th, 2016

Changed the filenames of all merged reads to better reflect the experiment (Changed all files with dss.feces to just dss)

Created a bash script to run trimmomatic, needed to create a fasta file containing appropriate adapter sequences.

Ran fastqc to check for sequence quality on the initial reads (R1 and R2), but could not view the generated html files.

History:

```
#Created Trimmomatic Directory and shell script
241  mkdir trimmomatic
242  cd trimmomatic
243  nano trimmomatic.dss.sh

#Renamed files generated by Flash2
244  cd ../merged.reads/
247  mv dss.feces.histogram dss.histogram
249  mv dss.feces.hist dss.hist
250  mv dss.feces.extendedFrags.fastq dss.extendedFrags.fastq
251  mv dss.feces.notCombined_1.fastq dss.notCombined_1.fastq
252  mv dss.feces.notCombined_2.fastq dss.notCombined_2.fastq
253  ls
254  cd ../trimmomatic/
256  nano trimmomatic.dss.sh

#Looked at the adapter fasta file used by Jill Hageu
266  cd /share/tearlab/Maga/Jill
267  ls
268  less adapter.fasta

#Edited the trimmomatic script
270  cd /share/magalab/Kat/DSS
271  cd trimmomatic/
272  nano trimmomatic.dss.sh
273  cd ../Other.files.from.genome.center/

#Ran FastQC on original fastq sequences
276  module load fastqc/v.0.11.2
277  module load fastqc/v0.11.2
278  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq -o /share/magalab/Kat/DSS/
279  ls
280  mkdir fastqc
281  mv DSS_S1_L001_R1_001_fastqc.zip fastqc/
282  ls
283  mv DSS_S1_L001_R1_001_fastqc.html fastqc/
284  ls
285  cd fastqc/
286  ls
287  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R1_002.fastq -o /share/magalab/Kat/DSS/fastqc
288  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq -o /share/magalab/Kat/DSS/fastqc
```

The partial trimmomatic script:

```
!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_$

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trimmomatic.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trimmomatic.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading trimmomatic v. 0.33"
module load trimmomatic/0.33

echo "Loading fastqc v. 0.11.2"
module load fastqc/v0.11.2

echo "Now running Trimmomatic on all merged reads."

time trimmomatic SE -threads 4 -phred33 /share/magalab/Kat/DSS/merged.reads/dss.extende$
```

### December 26th, 2016

Completed trimmomatic shell script, using TruSeq3 PE adapter fasta file provided by Trimmomatic. TruSeq3 adapters are used by MiSeq machines. Trimmomatic was run in SE mode, as reads are merged into one.

Completed Script:
```
#!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_sciences computer."

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trimmomatic.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trimmomatic.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading trimmomatic v. 0.33"
module load trimmomatic/0.33

echo "Loading fastqc v. 0.11.2"
module load fastqc/v0.11.2

echo "Now running Trimmomatic on all merged reads, removing TruSeq3 PE adapters (file provided by Trimmomatic)."

time trimmomatic SE -threads 4 -phred33 /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

echo "Trimming should be complete."
```

Also moved fastqc report files to local computer for viewing.

From local terminal:

```
scp -r klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat/DSS/fastqc //Users/kathleenfurtado/Documents/maga.lab/
```

### December 28, 2016
Checked quality of raw reads, merged reads, and trimmed reads. Realized trimmomatic was set up incorrectly (bad choice of quality score thresholds and minimum length), so reran trimmomatic with leading and trailing score thresholds of 30, sliding window of 4 with quality threshold of 30, minlen of 200.

Lost over 4 million reads with quality thresholds of 30, rerunning trimmomatic with leading and trailing thresholds of 25, sliding window of 4 with threshold quality 25. MILEN still 200.

**Initial FastQC reports:**

|Name|Number of Sequences|Poor Quality|Length|Fail|Warn|Notes|
|-----|:-------:|:-------------:|:-----------:|:--------------------------:|:----------------------:|:---------:|
|DSS_S1_L001_R1_001.fastq|16765099|0|251|Per base sequence content, Sequence Duplication levels, Overrepresented Sequences, Kmer Content|Per Sequence GC content||
|DSS_S1_L001_R2_001.fastq|16765099|0|251|Per base sequence content, Per base GC content, Sequence Duplication Levels, Overrepresented sequences, Kmer content|||
|dss.extendedFrags.fastq|16088672|0|251-492|Per tile sequence quality, Per base sequence content, Per sequence GC content, Sequence Duplication levels, Overrepresented Sequences, Kmer Content|Sequence Length Distribution||
|dss.extendedFrags.trimmed2.fastq|11684950|0|200-492|Per tile sequence quality, per base sequence content, per sequence GC content, Sequence Duplication levels, Overrepresented Sequences, Kmer content|Sequence Length Distribution|Quality thresholds of 30|
|dss.extendedFrags.trimmed.fastq|13590868|0|200-492|Per tile sequence quality, Per base sequence content, Per sequence GC content, Sequence Duplication Levels, Overrepresented Sequences|Sequence Length Distribution, Kmer content|Quality thresholds of 25|

History for today's session:
```
#Created new directories within FastQC directory
382  cd fastqc/
384  mkdir raw.reads
385  mv DSS* raw.reads/
386  ls
387  cd raw.reads/
388  ls
389  cd ..
390  mkdir merged
391  ls

#Ran FastQC on merged reads
392  module load fastqc/v0.11.2
403  fastqc /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq -o /share/magalab/Kat/DSS/fastqc/merged/
405  fastqc /share/magalab/Kat/DSS/merged.reads/dss.notCombined_1.fastq -o /share/magalab/Kat/DSS/fastqc/merged/
406  fastqc /share/magalab/Kat/DSS/merged.reads/dss.notCombined_2.fastq -o /share/magalab/Kat/DSS/fastqc/merged/

#Created new directory within fastqc/ for trimmed reads
408  mkdir trimmed
409  cd trimmed

#Reran fastqc
410  fastqc /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq -o /share/magalab/Kat/DSS/fastqc/trimmed/

#Renamed previous trimmed files prior to rerunning trimmomatic
411  cd ../../trimmomatic/
413  mv dss.extendedFrags.trimmed.fastq dss.extendedFrags.trimmed1.fastq
414  ls

#Edited bash script to change quality score thresholds for trimmomatic (30)
415  nano trimmomatic.dss.sh
416  ./trimmomatic.dss.sh
#input: 16088672, surviving: 11684950 (72.63%), dropped: 4403722(27.37%)

#Renamed fastqc report names prior to rerunning fastqc on newly trimmed reads.
417  cd ../fastqc/trimmed/
418  ls
419  mv dss.extendedFrags.trimmed_fastqc.html dss.extendedFrags.trimmed1.html
420  mv dss.extendedFrags.trimmed_fastqc.zip dss.extendedFrags.trimmed1.zip

#Reran fastqc
422  fastqc /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq -o /share/magalab/Kat/DSS/fastqc/trimmed/

#Needed to run trimmomatic a third time. Changed names of trimmed files and of fastqc reports again.
423  mv dss.extendedFrags.trimmed_fastqc.html dss.extendedFrags.trimmed2.html
424  mv dss.extendedFrags.trimmed_fastqc.zip dss.extendedFrags.trimmed2.zip
425  ls
426  cd ../../trimmomatic/
427  ls
428  mv dss.extendedFrags.trimmed.fastq dss.extendedFrags.trimmed2.fastq

#Edited bash script to change quality score thresholds for trimmomatic (25)
429  nano trimmomatic.dss.sh
430  ./trimmomatic.dss.sh

#Reran trimmomatic on newly trimmed reads.
431  fastqc /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq -o /share/magalab/Kat/DSS/fastqc/trimmed/
#input: 16088672, surviving: 13590868 (84.47%), dropped: 2497804 (15.53%)
```

Final Bash Script used for trimmomatic:
```
#!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_sciences computer."

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trimmomatic.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trimmomatic.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading trimmomatic v. 0.33"
module load trimmomatic/0.33

echo "Loading fastqc v. 0.11.2"
module load fastqc/v0.11.2

echo "Now running Trimmomatic on all merged reads, removing TruSeq3 PE adapters (file provided by Trimmomatic)."

time trimmomatic SE -threads 4 -phred33 /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:25 TRAILING:25 SLIDINGWINDOW:4:25 MINLEN:200

echo "Trimming should be complete."
```

### December 30, 2016

Ran bowtie2-build to build indexes of Pig mitochondrial and PhiX phage genomes.

```
#Created directory for index building and screening and created script to run bowtie2 build.
436  mkdir index.screen
437  ls
438  cd index.screen/
439  nano dss.bowtie2.sh
443  chmod u+x dss.bowtie2.sh
444  ./dss.bowtie2.sh

#Changed permissions on genome files to allow usage by bowtie2
456  chmod a+rwx sus.scrofa_mitochondria_genome.fa
457  chmod a+rwx phi.x_genome.fna
459  ls -lah
460  ./dss.bowtie2.sh
461  nano dss.bowtie2.sh
462  ls

#Created directories for output and error
463  mkdir fastq_screen.out
464  mkdir fastq_screen.err
465  nano dss.bowtie2.sh
466  ./dss.bowtie2.sh

#Double checked that genome files were actually in FASTA format.
476  cat phi.x_genome.fna
477  cat sus.scrofa_mitochondria_genome.fa

#changed permissions again on genome files to try to get bowtie2 to run
489  chmod g+s phi.x_genome.fna
490  chmod g+s sus.scrofa_mitochondria_genome.fa

#Renamed genome files
500  mv phi_x_genome.fna phi.x_genome.fasta
501  mv sus_scrofa_mito_genome.fa sus.scrofa_mito_genome.fasta
502  ls

#Edited bowtie2 script to allow for usage of files in current directory (specifying the path was not working.)
505  nano dss.bowtie2.sh
506  ./dss.bowtie2.sh
507  ls
#IT FINALLY WORKED.

#Created bash script for running fastq_screen
509  module avail
510  nano fastq.screen.sh
```

Completed bowtie2 bash script. To run, genome files must be present in the correct (current) directory.

```
#!/bin/bash
##

echo "Setting up output and error directories for bowtie2 and fastq_screen."

#SBATCH -o /share/magalab/Kat/DSS/index.screen/fastq_screen.out
#SBATCH -e /share/magalab/Kat/DSS/index.screen/fastq_screen.err

echo "Loading bowtie2 to build indexes."

module load bowtie2/2.2.8

echo "Building index using phage genome (Sequence from NCBI Genomes database)."

time bowtie2-build -f ./phi.x_genome.fasta phi.x.174

echo "Building index using pig mitochondrial genome (Sequence from NCBI Genomes database)."

time bowtie2-build -f ./sus.scrofa_mito_genome.fasta pig.mt

echo "Indexing should be complete. 6 basefiles should be created for each genome."
```

### January 8, 2017

Installed, configured, and ran fastq_screen.

fastq_screen requires a .conf file for configuration. Below is the .conf file used for the DSS study (adapted from the .conf example which may be downloaded from the fastq_screen webpage). Note: only the portions which were edited have been copied here.

```
# This is an example configuration file for FastQ Screen

############################
## Bowtie, Bowtie 2 or BWA #
############################
## If the Bowtie, Bowtie 2 or BWA binary is not in your PATH, you can set
## this value to tell the program where to find your chosen aligner.  Uncomment
## the relevant line below and set the appropriate location.  Please note,
## this path should INCLUDE the executable filename.

#BOWTIE /usr/local/bin/bowtie/bowtie
BOWTIE2 /software/modules/3.2.10/x86_64-linux-ubuntu14.04/Modules/3.2.10/modulefiles bowtie2/2.2.8
#BWA /usr/local/bwa/bwa


##############
## DATABASES #
##############
## This section enables you to configure multiple genomes databases (aligner index
## files) to search against in your screen.  For each genome you need to provide a
## database name (which can't contain spaces) and the location of the aligner index
## files.
##
## The path to the index files SHOULD INCLUDE THE BASENAME of the index, e.g:
## /data/public/Genomes/Human_Bowtie/GRCh37/Homo_sapiens.GRCh37
## Thus, the index files (Homo_sapiens.GRCh37.1.bt2, Homo_sapiens.GRCh37.2.bt2, etc.)
## are found in a folder named 'GRCh37'.
##
## If, for example, the Bowtie, Bowtie2 and BWA indices of a given genome reside in
## the SAME FOLDER, a SINLGE path may be provided to ALL the of indices.  The index
## used will be the one compatible with the chosen aligner (as specified using the
## --aligner flag).
##
## The entries shown below are only suggested examples, you can add as many DATABASE
## sections as required, and you can comment out or remove as many of the existing
## entries as desired.  We suggest including genomes and sequences that may be sources
## of contamination either because they where run on your sequencer previously, or may
## have contaminated your sample during the library preparation step.
##
## Human - sequences available from
## ftp://ftp.ensembl.org/pub/current/fasta/homo_sapiens/dna/
#DATABASE       Human   /data/public/Genomes/Human_Bowtie/GRCh37/Homo_sapiens.GRCh37
##
## Mouse - sequence available from
## ftp://ftp.ensembl.org/pub/current/fasta/mus_musculus/dna/
#DATABASE       Mouse   /data/public/Genomes/Mouse/NCBIM37/Mus_musculus.NCBIM37
##
## Ecoli- sequence available from EMBL accession U00096.2
#DATABASE       Ecoli   /data/public/Genomes/Ecoli/Ecoli
##
## PhiX - sequence available from Refseq accession NC_001422.1
#DATABASE       PhiX    /data/public/Genomes/PhiX/phi_plus_SNPs
##
## Adapters - sequence derived from the FastQC contaminats file found at: www.bioinformatics.babr$
#DATABASE       Adapters        /data/public/Genomes/Contaminants/Contaminants
##
## Vector - Sequence taken from the UniVec database
## http://www.ncbi.nlm.nih.gov/VecScreen/UniVec.html
#DATABASE       Vectors         /data/public/Genomes/Vectors/Vectors
##
## Pig Mitochondria
DATABASE pig.mt /share/magalab/Kat/DSS/index.screen/pig.mt/pig.mt
##
## PhiX 174
DATABASE phi.x.174 /share/magalab/Kat/DSS/index.screen/phi.x.174/phi.x.174
```

Script used to run fastq_screen:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/index.build/fastq_screen.out
#SBATCH -e /share/magalab/Kat/DSS/index.build/fastq_screen.err

echo "Loading bowtie2 v. 2.2.8"

module load bowtie2/2.2.8

echo "Running fastq_screen to check merged and trimmed reads for hits within PhiX and Sus Scrofa mitochondrial genome."

time fastq_screen_v0.9.5/fastq_screen --threads 8 --subset 0 --aligner bowtie2 --force --nohits /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq

echo "Reads should be screened."
echo "Running FastQC to check new read quality."

module load fastqc/v0.11.2

time fastqc /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq -o /share/magalab/Kat/DSS/index.screen

echo "FastQC is complete. Reports are located in the index.screen/ directory."
```

History from today's session with details on installation of fastq_screen

```
#Installed fastq_screen and test dataset
cd index.screen/
398  ls
399  wget http://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/fastq_screen_v0.9.5.tar.gz
400  wget http://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/fastq_screen_test_dataset.tar.gz
401  ls
#unzip files
402  tar xvzf fastq_screen_v0.9.5.tar.gz
403  tar xvzf fastq_screen_test_dataset.tar.gz
404  cd fastq_screen_v0.9.5/
405  ls

#copy and edit .conf file to set aligner (bowtie2) and path.
406  cp fastq_screen.conf.example fastq_screen.conf
407  nano fastq_screen.conf

#tried testing fastq_screen on test dataset, but could not configure. Moved on to editing script and .conf file for DSS dataset.

#created directories for bowtie-2 indices for retrieval by fastq_screen
476  mkdir phi.x.174
477  mkdir pig.mt
478  mv phi.x* phi.x.174/
479  ls
480  cd phi.x.174
481  ls
482  cd ..
483  mv pig.mt* pig.mt/
485  cd pig.mt
486  ls

#edited .conf file to adjust path for indices.
497  cd fastq_screen_v0.9.5/
498  nano fastq_screen.conf
500  ./dss.fastq.screen.sh
```

January 19th, 2017

After rerunning fastq_screen and looking at the html report, it appears that there is 0% contamination. Due to this highly unlikely result, there may have been an issue with the initial genome database construction or with some part of the fastq_screen configuration.
