# DSS/DSS-hLZ Workflow and History

### Logging into Cabernet server:

```ssh klfurtado@cabernet.genomecenter.ucdavis.edu```

Password: **Ydm.M6.z**

If personal account is not working, use:

```ssh eamaga@cabernet.genomecenter.ucdavis.edu```

Will be prompted to enter password.

### October 7th, 2016

Loaded files into the server from local hard drive.

From terminal on local computer:

```
165  scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS/ eamaga@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat
166  scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS-hlZ/ eamaga@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat
```

### October 21, 2016

Logged in using personal username and password:

``` ssh klfurtado@cabernet.genomecenter.ucdavis.edu```

Prompted to enter password: Ydm.M6.z

Change into directory:

```cd share/tearlab/Maga/Kat```

*Permission denied for the following commands:*

Within server: ```mkdir```

From local terminal:
```
scp -r //Users/kathleenfurtado/Documents/maga.lab/dss/20150401_dss_mapping.file_contents.txt klfurtado@cabernet.genomecenter.ucdavis.edu:/share/tearlab/Maga/Kat/DSS
```

### December 9th, 2016

Added sequence files to new genome center server directory.

```
ssh klfurtado@cabernet.genomecenter.ucdavis.edu
scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS/ klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat
scp -r //Users/kathleenfurtado/Documents/maga.lab/illumina.seqs/DSS-hLZ/ klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat
```

### December 12, 2016

Created "Practice" directory for writing bash scripts, changed permissions, and executed command within same directory using ./

```
162  cd Kat
163  mkdir Practice
165  cd Practice
166  nano
167  nano awesome.sh
168  chmod u+x awesome.sh
169  ls -lah awesome.sh
173  ./awesome.sh
```

Unzipped fastq.gz files for DSS study. DSS-hLZ files were not unzipped (broken pipe?)

### December 13, 2016

Uploaded Mapping Files for DSS-hLZ. Unzipped DSS-hLZ data files.

### December 14, 2016

Wrote shell script to run flash2 and merge forward and reverse reads for DSS study.

```
212  mkdir merged.reads
213  cd merged.reads
214  nano flash2.dss.sh
215  nano flash2.dss.sh
216  ./flash2.dss.sh
217  chmod u+x flash2.dss.sh
218  ./flash2.dss.sh
219  cd /share/magalab/Kat
220  ls
221  cd DSS
222  ls
223  cd merged.reads
224  ls
225  less dss.feces.histogam
226  less dss.feces.histogram
```

Script: flash2.dss.sh

```
#!/bin/bash
##

echo "Setting up output and error directories"
#SBATCH -o /share/magalab/Kat/DSS/merged.reads/flash2.out
#SBATCG -e /share/magalab/Kat/DSS/merged.reads/flash2.err

echo "Loading Flash2 module"
module load flash2/c41a82e

#Average read length should be 250 bp, average fragment length should be between 290 and 300, and quality shou$

echo "Options -r 250, -f 290, and -s 29 are used to calculate max overlap"

time flash2 /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq -o$

echo "Merging should be complete"
```

Allowed to run until December 15, received error:

```
packet_write_wait: Connection to 128.120.143.100 port 22: Broken pipe
```

Several output files were still created.

### December 19th, 2016

Changed the filenames of all merged reads to better reflect the experiment (Changed all files with dss.feces to just dss)

Created a bash script to run trimmomatic, needed to create a fasta file containing appropriate adapter sequences.

Ran fastqc to check for sequence quality on the initial reads (R1 and R2), but could not view the generated html files.

History:

```
#Created Trimmomatic Directory and shell script
241  mkdir trimmomatic
242  cd trimmomatic
243  nano trimmomatic.dss.sh

#Renamed files generated by Flash2
244  cd ../merged.reads/
247  mv dss.feces.histogram dss.histogram
249  mv dss.feces.hist dss.hist
250  mv dss.feces.extendedFrags.fastq dss.extendedFrags.fastq
251  mv dss.feces.notCombined_1.fastq dss.notCombined_1.fastq
252  mv dss.feces.notCombined_2.fastq dss.notCombined_2.fastq
253  ls
254  cd ../trimmomatic/
256  nano trimmomatic.dss.sh

#Looked at the adapter fasta file used by Jill Hageu
266  cd /share/tearlab/Maga/Jill
267  ls
268  less adapter.fasta

#Edited the trimmomatic script
270  cd /share/magalab/Kat/DSS
271  cd trimmomatic/
272  nano trimmomatic.dss.sh
273  cd ../Other.files.from.genome.center/

#Ran FastQC on original fastq sequences
276  module load fastqc/v.0.11.2
277  module load fastqc/v0.11.2
278  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq -o /share/magalab/Kat/DSS/
279  ls
280  mkdir fastqc
281  mv DSS_S1_L001_R1_001_fastqc.zip fastqc/
282  ls
283  mv DSS_S1_L001_R1_001_fastqc.html fastqc/
284  ls
285  cd fastqc/
286  ls
287  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R1_002.fastq -o /share/magalab/Kat/DSS/fastqc
288  time fastqc /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq -o /share/magalab/Kat/DSS/fastqc
```

The partial trimmomatic script:

```
!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_$

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trimmomatic.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trimmomatic.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading trimmomatic v. 0.33"
module load trimmomatic/0.33

echo "Loading fastqc v. 0.11.2"
module load fastqc/v0.11.2

echo "Now running Trimmomatic on all merged reads."

time trimmomatic SE -threads 4 -phred33 /share/magalab/Kat/DSS/merged.reads/dss.extende$
```

### December 26th, 2016

Completed trimmomatic shell script, using TruSeq3 PE adapter fasta file provided by Trimmomatic. TruSeq3 adapters are used by MiSeq machines. Trimmomatic was run in SE mode, as reads are merged into one.

Completed Script:
```
#!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_sciences computer."

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trimmomatic.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trimmomatic.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading trimmomatic v. 0.33"
module load trimmomatic/0.33

echo "Loading fastqc v. 0.11.2"
module load fastqc/v0.11.2

echo "Now running Trimmomatic on all merged reads, removing TruSeq3 PE adapters (file provided by Trimmomatic)."

time trimmomatic SE -threads 4 -phred33 /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.fastq ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

echo "Trimming should be complete."
```

Also moved fastqc report files to local computer for viewing.

From local terminal:

```
scp -r klfurtado@cabernet.genomecenter.ucdavis.edu:/share/magalab/Kat/DSS/fastqc //Users/kathleenfurtado/Documents/maga.lab/
```

### December 28, 2016
Checked quality of raw reads, merged reads, and trimmed reads. Realized trimmomatic was set up incorrectly (bad choice of quality score thresholds and minimum length), so reran trimmomatic with leading and trailing score thresholds of 30, sliding window of 4 with quality threshold of 30, minlen of 200.

Lost over 4 million reads with quality thresholds of 30, rerunning trimmomatic with leading and trailing thresholds of 25, sliding window of 4 with threshold quality 25. MILEN still 200.

**Initial FastQC reports:**

|Name|Number of Sequences|Poor Quality|Length|Fail|Warn|Notes|
|-----|:-------:|:-------------:|:-----------:|:--------------------------:|:----------------------:|:---------:|
|DSS_S1_L001_R1_001.fastq|16765099|0|251|Per base sequence content, Sequence Duplication levels, Overrepresented Sequences, Kmer Content|Per Sequence GC content||
|DSS_S1_L001_R2_001.fastq|16765099|0|251|Per base sequence content, Per base GC content, Sequence Duplication Levels, Overrepresented sequences, Kmer content|||
|dss.extendedFrags.fastq|16088672|0|251-492|Per tile sequence quality, Per base sequence content, Per sequence GC content, Sequence Duplication levels, Overrepresented Sequences, Kmer Content|Sequence Length Distribution||
|dss.extendedFrags.trimmed2.fastq|11684950|0|200-492|Per tile sequence quality, per base sequence content, per sequence GC content, Sequence Duplication levels, Overrepresented Sequences, Kmer content|Sequence Length Distribution|Quality thresholds of 30|
|dss.extendedFrags.trimmed.fastq|13590868|0|200-492|Per tile sequence quality, Per base sequence content, Per sequence GC content, Sequence Duplication Levels, Overrepresented Sequences|Sequence Length Distribution, Kmer content|Quality thresholds of 25|
