# DSS/DSS-hLZ Workflow and History (Part 2)

### February 16, 2017

**Fastq_screen troubleshooting**

Confirmed that bowtie2-build did indeed build the correct indexes, by running ```bowtie2-inspect``` within the directories where the bowtie2 indexes are kept (pig.mt/ and phi.x.174/ )

Usage:

```
bowtie2-inspect -v pig.mt
bowtie2-inspect -v phi.x.174
```

### February 17, 2017

**Fastq_screen troubleshooting**

Tried adjusting the filepath in the configuration file for bowtie2 aligner. Nothing else in the .conf file was changed.

New .conf file changes:

```
# This is an example configuration file for FastQ Screen

############################
## Bowtie, Bowtie 2 or BWA #
############################
## If the Bowtie, Bowtie 2 or BWA binary is not in your PATH, you can set
## this value to tell the program where to find your chosen aligner.  Uncomment
## the relevant line below and set the appropriate location.  Please note,
## this path should INCLUDE the executable filename.

#BOWTIE	/usr/local/bin/bowtie/bowtie
BOWTIE2 module load bowtie2/2.2.8
#BWA /usr/local/bwa/bwa
```

Edited ```dss.fastq.screen.sh``` script to remove FastQC steps until fastq_screen is completely fixed.

### March 3, 2017

Was finally able to complete a successful run for Fastq_screen, by editing the path to bowtie2 in the .conf file. Below is the final edits and result:

```
# This is an example configuration file for FastQ Screen

############################
## Bowtie, Bowtie 2 or BWA #
############################
## If the Bowtie, Bowtie 2 or BWA binary is not in your PATH, you can set
## this value to tell the program where to find your chosen aligner.  Uncomment
## the relevant line below and set the appropriate location.  Please note,
## this path should INCLUDE the executable filename.

#BOWTIE	/usr/local/bin/bowtie/bowtie
BOWTIE2 /software/bowtie2/2.2.8/x86_64-linux-ubuntu14.04/bin/bowtie2
#BWA /usr/local/bwa/bwa
```

**Results:**

#Fastq_screen version: 0.9.5

#Aligner: bowtie2

#Processing all reads in FASTQ files

| Genome | #Reads_processed	| #Unmapped	| %Unmapped | #One_hit_one_genome	| %One_hit_one_genome	| #Multiple_hits_one_genome |	%Multiple_hits_one_genome	| #One_hit_multiple_genomes | 	%One_hit_multiple_genomes | Multiple_hits_multiple_genomes | 	%Multiple_hits_multiple_genomes |
|:----|:----|:------|:-----|:-----|:------|:------|:-----|:-----|:-----|:-----|:-----|
| pig.mt | 13590868	| 13590867 | 100.00 | 1	| 0.00 | 0 | 0.00	| 0	| 0.00	| 0	| 0.00 |
| phi.x.174	| 13590868 |	12777036 |	94.01 |	773371 |	5.69 |	40461 |	0.30 |	0 |	0.00 |	00.00 | 0.00 |

%Hit_no_genomes: 94.01
