# DSS/DSS-hLZ Workflow and History (Part 2)

### February 16, 2017

**Fastq_screen troubleshooting**

Confirmed that bowtie2-build did indeed build the correct indexes, by running ```bowtie2-inspect``` within the directories where the bowtie2 indexes are kept (pig.mt/ and phi.x.174/ )

Usage:

```
bowtie2-inspect -v pig.mt
bowtie2-inspect -v phi.x.174
```

### February 17, 2017

**Fastq_screen troubleshooting**

Tried adjusting the filepath in the configuration file for bowtie2 aligner. Nothing else in the .conf file was changed.

New .conf file changes:

```
# This is an example configuration file for FastQ Screen

############################
## Bowtie, Bowtie 2 or BWA #
############################
## If the Bowtie, Bowtie 2 or BWA binary is not in your PATH, you can set
## this value to tell the program where to find your chosen aligner.  Uncomment
## the relevant line below and set the appropriate location.  Please note,
## this path should INCLUDE the executable filename.

#BOWTIE	/usr/local/bin/bowtie/bowtie
BOWTIE2 module load bowtie2/2.2.8
#BWA /usr/local/bwa/bwa
```

Edited ```dss.fastq.screen.sh``` script to remove FastQC steps until fastq_screen is completely fixed.

### March 3, 2017

Was finally able to complete a successful run for Fastq_screen, by editing the path to bowtie2 in the .conf file. Below is the final edits and result:

```
# This is an example configuration file for FastQ Screen

############################
## Bowtie, Bowtie 2 or BWA #
############################
## If the Bowtie, Bowtie 2 or BWA binary is not in your PATH, you can set
## this value to tell the program where to find your chosen aligner.  Uncomment
## the relevant line below and set the appropriate location.  Please note,
## this path should INCLUDE the executable filename.

#BOWTIE	/usr/local/bin/bowtie/bowtie
BOWTIE2 /software/bowtie2/2.2.8/x86_64-linux-ubuntu14.04/bin/bowtie2
#BWA /usr/local/bwa/bwa
```

**Results:**

#Fastq_screen version: 0.9.5

#Aligner: bowtie2

#Processing all reads in FASTQ files

| Genome | #Reads_processed	| #Unmapped	| %Unmapped | #One_hit_one_genome	| %One_hit_one_genome	| #Multiple_hits_one_genome |	%Multiple_hits_one_genome	| #One_hit_multiple_genomes | 	%One_hit_multiple_genomes | Multiple_hits_multiple_genomes | 	%Multiple_hits_multiple_genomes |
|:----|:----|:------|:-----|:-----|:------|:------|:-----|:-----|:-----|:-----|:-----|
| pig.mt | 13590868	| 13590867 | 100.00 | 1	| 0.00 | 0 | 0.00	| 0	| 0.00	| 0	| 0.00 |
| phi.x.174	| 13590868 |	12777036 |	94.01 |	773371 |	5.69 |	40461 |	0.30 |	0 |	0.00 |	00.00 | 0.00 |

%Hit_no_genomes: 94.01

Ran trimmomatic again to make sure TruSeq3 adapters were actually removed (received an error previously which made me question whether the adapters were actually found by trimmomatic). Copied sequences from timflutre/trimmomatic on Github to find TruSeq3 adapters previously used with Trimmomatic.

Created a personal adapter file for the server:

truseq3.adapters.fa

```
>PrefixPE/1
TACACTCTTTCCCTACACGACGCTCTTCCGATCT
>PrefixPE/2
GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT
>PE1
TACACTCTTTCCCTACACGACGCTCTTCCGATCT
>PE1_rc
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA
>PE2
GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT
>PE2_rc
AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC
```

Did not receive any errors when starting trimmomatic the second time.

### March 7, 2017

Results from trimmomatic:

```
Module trimmomatic-0.33-static loaded.
TrimmomaticSE: Started with arguments: -threads 4 -phred33 /share/magalab/Kat/DSS/mothur/dss.extendedFrags.trimmed.tagged_filter.fastq /share/magalab/Kat/DSS/trimmomatic/dss.extendedFrags.trimmed.tagged_filter.trimmed.fastq ILLUMINACLIP:truseq3.adapters.fa:2:30:10 LEADING:25 TRAILING:25 SLIDINGWINDOW:4:25 MINLEN:200
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA'
Using Long Clipping Sequence: 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC'
Using Long Clipping Sequence: 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
Using Long Clipping Sequence: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT'
ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Input Reads: 12777035 Surviving: 12777027 (100.00%) Dropped: 8 (0.00%)
TrimmomaticSE: Completed successfully
```

Renamed fastq file that has been retrimmed and filtered, for easier use. Placed new file in the mothur directory. New file name: ```dss.merged.trimmed.screened.fastq```

**Proceeding on to working with mothur.**

Must convert fastq to fasta with ```fastq.info```

Must edit normal mothur syntax for running in linux shell.

Script:
```
#!/bin/bash
##

echo "Setting up output and error directories, running on 4 nodes, partition to animal_sciences computer."

#SBATCH -o /share/magalab/Kat/DSS/mothur/mothur.out
#SBATCH -e /share/magalab/Kat/DSS/mothur/mothur.err
#SBATCH -n 4
#SBATCH -p animal_sciences

echo "Loading mothur v. 1.36.1 from server."

module load mothur/1.36.1

echo "Generating FASTA file from merged, trimmed, filtered FASTQ."

mothur -q "#fastq.info(fastq=dss.merged.trimmed.screened.fastq, qfile=T, format=illumina1.8+);"

echo "FASTA file and quality file should have been generated."
echo "Next step will be to use summary.seqs."
```

### March 8, 2017
**Continuing to work with mothur**

Use ```summary.seqs``` to build a distribution, then use ```screen.seqs``` to remove the longest (poorest quality/chimeric) reads.

### March 9, 2017
**Mothur**

```fastq.info``` finished, started running ```summary.seqs``` with on new fasta file, "dss.merged.trimmed.screen.fasta"

Script: dss.mothur_summary.seqs.sh
```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/mothur/mothur.out
#SBATCH -e /share/magalab/Kat/DSS/mothur/mothur.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 2-00:00:00

echo "Setting up output and error directories, running on 4 nodes, partition to animal_sciences computer."

echo "Loading mothur"

module load mothur/1.36.1

echo "Creating a summary of sequence length distribution."

mothur -q "#summary.seqs(fasta=dss.merged.trimmed.screened.fasta);"

echo "Check summary, then run screen.seqs."
```
