# DSS/DSS-hLZ Workflow and History (Part 3)

### June 7, 2017

**Working with QIIME**

Tried to validate mapping files by:

Loading QIIME: ```module load qiime/1.9.1```

Running: ```validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -o /share/magalab/Kat/DSS/qiime/feces.mapping_output```

Received following error:

```
(validate_mapping_file.py:30957): Gdk-CRITICAL **: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed
Traceback (most recent call last):
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/bin/validate_mapping_file.py", line 14, in <module>
    from qiime.util import parse_command_line_parameters, get_options_lookup,\
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/qiime/util.py", line 49, in <module>
    from skbio.util import remove_files, create_dir
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/__init__.py", line 15, in <module>
    import skbio.io
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/io/__init__.py", line 309, in <module>
    import_module('skbio.io.clustal')
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/io/clustal.py", line 123, in <module>
    from skbio.alignment import Alignment
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/alignment/__init__.py", line 230, in <module>
    from ._alignment import Alignment, SequenceCollection, StockholmAlignment
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/alignment/_alignment.py", line 21, in <module>
    from skbio.stats.distance import DistanceMatrix
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/stats/distance/__init__.py", line 193, in <module>
    from ._base import (DissimilarityMatrixError, DistanceMatrixError,
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/stats/distance/_base.py", line 16, in <module>
    import matplotlib.pyplot as plt
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.py", line 109, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/backend_gtk3agg.py", line 11, in <module>
    from . import backend_gtk3
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/backend_gtk3.py", line 55, in <module>
    cursors.MOVE          : Gdk.Cursor.new(Gdk.CursorType.FLEUR),
TypeError: constructor returned NULL
```

### June 8, 2017

After contacting Genome Center, error was due to QIIME attempting to run matplotlib in sbatch. Needed to change the ```matplotlibrc``` file, specifically altering the backend option, to 'Agg'.

To do this, visited:
http://matplotlib.org/users/customizing.html

Scrolled to bottom of page and copied link to sample matplotlibrc file. In terminal, within the directory where I wanted to download the file, used ```wget``` to download file from the webpage. Then, altered ONLY the backend option to 'Agg' using ```nano```.

As default, QIIME will look for ```matplotlibrc``` file in the closest possible place. If running from sbatch files from a particular directory, place the edited ```matplotlibrc``` file there, and QIIME will look there first and find the appropriate, edited file.

Portion of ```matplotlibrc``` changed:

```
#### CONFIGURATION BEGINS HERE

# The default backend; one of GTK GTKAgg GTKCairo GTK3Agg GTK3Cairo
# MacOSX Qt4Agg Qt5Agg TkAgg WX WXAgg Agg Cairo GDK PS PDF SVG
# Template.
# You can also deploy your own backend outside of matplotlib by
# referring to the module name (which must be in the PYTHONPATH) as
# 'module://my_backend'.
backend      : Agg

```

QIIME now functions normally.

### June 9, 2017

After copying and pasting the older version of the DSS contents mapping file into a new Excel spreadsheet and saving as a Tab delimited text (.txt) file, was able to validate mapping files with no issues.

Script for validating the mapping files for both feces and contents sample sets:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/validate.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/validate.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Validating Mapping Files"

time validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -o /share/magalab/Kat/DSS/qiime/validate.mapping

time validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -o /share/magalab/Kat/DSS/qiime/validate.mapping

echo "Check log files"
```

Next step is to demultiplex. Scripts for demultiplexing for both feces and contents sample sets:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/demultiplex.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/demultiplex.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Running extract_barcodes.py in order to use split_libraries_fastq.py later on."

time extract_barcodes.py -f /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.feces_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/qiime/mapping.file/MappingFiles_DSS_feces.txt -a

time extract_barcodes.py -f /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.contents_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -a

echo "Running split_libraries_fastq.py to demultiplex."

time split_libraries_fastq.py  -i /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.feces_split_library_output -b /share/magalab/Kat/DSS/qiime/dss.feces_barcodes/barcodes.fastq -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -q 34 --phred_offset 33

time split_libraries_fastq.py  -i /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.contents_split_library_output -b /share/magalab/Kat/DSS/qiime/dss.contents_barcodes/barcodes.fastq -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -q 34 --phred_offset 33

echo "Demultiplexed files ready for Chimera detection."
```

Notes about options:

For ```extract_barcodes.py```

 * **-c barcode_single_end** indicates that input was a single fastq file, with the barcode located at the beginning of each read.
 * **-l 8** indicates that barcodes are 8 bp long.
 * **-a** (--attempt_read_reorientation) "Will attempt to search for the forward and reverse primer in the read and adjust the sequence orientation to match the orientation of the forward primer. An exact match for the forward and reverse complemented versions of the primers are tested for, and sequences are reverse complemented, if necessary, before writing. Sequences without an exact match are written to a separate output fastq file, labeled as _no_primer_match.fastq"

For ```split_libraries_fastq.py```

 * **-b filepath** indicates path to barcodes.fastq file, generated from ```extract_barcodes.py```
 * ** -q 34** indicates maximum UNACCEPTABLE Phred quality score. QIIME will only accept reads with Q 35 or higher. Value arbitrarily decided based on FASTQC output for ```dss.merged.trimmed.screened.good.fastq```.
 * ** --phred_offset 33** specifies the ascii offset for decoding phred scores.
