# DSS and DSS-hLZ Workflow (NEW!)

### July 6 2017

** Merging Reads**

Merged reads from DSS project again, to ensure that they were merged correctly and accurately.

Used Flash2 to merge

Script:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.reads$ cat flash2.dss.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.reads/merge.out
#SBATCH -e /share/magalab/Kat/DSS/merged.reads/merge.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading Flash2 module"
module load flash2/c41a82e

echo "Average read length should be 250 bp, average fragment length should be between 290 and 300, and quality threshold above 29."

echo "Options -r 250, -f 290, and -s 29 are used to calculate max overlap"

time flash2 /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq -o dss -r 250 -f 290 -s 29

echo "Merging complete"
```

Output:
```
[FLASH] Read combination statistics:
[FLASH]     Total pairs:       16765099
[FLASH]     Discarded pairs:   0
[FLASH]     Percent Discarded: 0.00%
[FLASH]     Combined pairs:    16088672
[FLASH]     Uncombined pairs:  676427
[FLASH]     Percent combined:  95.97%
[FLASH]
[FLASH] Writing histogram files.
[FLASH]
[FLASH] FLASH v2.2.00 complete!
[FLASH] 654.203 seconds elapsed
Merging complete
```

Histogram:
```
251	2647
252	2385
253	2201
254	1908
255	1867
256	2350
257	3161
258	3500
259	3810
260	4031
261	3122
262	3868
263	4510
264	4361
265	6027
266	7353
267	7264
268	6615
269	8238
270	8097
271	9334
272	14780
273	15617
274	15615
275	15969
276	15646
277	16875
278	21073
279	45198
280	39336
281	75555
282	193544
283	105747
284	93975
285	116800
286	150090
287	185202
288	132648
289	129615
290	114477
291	150406
292	242799
293	122430
294	129445
295	112009
296	109547
297	122412
298	131117
299	178497
300	302348
301	2579643
302	9138339
303	197663
304	10115
305	3711
306	2150
307	1003
308	881
309	851
310	864
311	782
312	891
313	871
314	964
315	1013
316	1067
317	1186
318	1193
319	1273
320	1340
321	1492
322	1605
323	1691
324	1989
325	2091
326	2318
327	2526
328	2886
329	3146
330	3559
331	4081
332	4276
333	5004
334	5459
335	6021
336	6801
337	7788
338	8656
339	8274
340	8330
341	8201
342	8334
343	8581
344	9101
345	9328
346	9645
347	9924
348	10197
349	10410
350	10526
351	10752
352	11085
353	11140
354	11179
355	11150
356	11308
357	11289
358	11635
359	11154
360	11130
361	11274
362	11347
363	11117
364	10912
365	10908
366	10688
367	10755
368	10489
369	10354
370	10371
371	10226
372	10053
373	10055
374	9977
375	9713
376	9468
377	9522
378	9427
379	9300
380	9298
381	9186
382	8980
383	8861
384	8709
385	8823
386	8604
387	8706
388	8338
389	8364
390	8247
391	7790
392	8014
393	8067
394	7840
395	7457
396	7689
397	7257
398	7322
399	7308
400	7284
401	7154
402	6941
403	6933
404	6897
405	6893
406	6840
407	6495
408	6415
409	6453
410	6504
411	6274
412	6266
413	6078
414	5986
415	5959
416	5794
417	5680
418	5711
419	5593
420	5491
421	5441
422	5312
423	5263
424	5288
425	5095
426	5217
427	4976
428	4761
429	4836
430	4869
431	4599
432	4614
433	4491
434	4379
435	4352
436	4166
437	4135
438	4023
439	3880
440	3775
441	3743
442	3571
443	3439
444	3278
445	3221
446	3066
447	2951
448	2791
449	2591
450	2588
451	2358
452	2248
453	2155
454	1966
455	1824
456	1803
457	1651
458	1534
459	1417
460	1305
461	1235
462	1171
463	1048
464	908
465	927
466	810
467	714
468	746
469	630
470	604
471	521
472	447
473	502
474	401
475	373
476	430
477	392
478	335
479	278
480	288
481	235
482	292
483	245
484	241
485	228
486	262
487	237
488	262
489	272
490	342
491	225
492	359
```

Read lengths 282-303 appear to be most frequent.

**Converting Merged FASTQ to FASTA and qual file for Demultiplexing**

Using QIIME script ```convert_fastaqual_fastq.py```

Script:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.fastaqual/convert.out
#SBATCH -e /share/magalab/Kat/DSS/merged.fastaqual/convert.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Converting Merged reads to FASTA plus a quality file for demultiplexing."

time convert_fastaqual_fastq.py -c fastq_to_fastaqual -f /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq -o /share/magalab/Kat/DSS/merged.fastaqual

echo "Done converting."
```

July 7 2017

Sequences were successfully transformed into FASTA format and a qual file was created.

For demultiplexing, I added an additional column to the mapping files for "ReversePrimer" and validated the mapping files using ```validate_mapping_file.py```

First Lines of Mapping Files:

```
#SampleID	BarcodeSequence	LinkerPrimerSequence	ReversePrimer	Trial	Time	TrialTime	Description
#Analysis of the Bacteria in Intestinal Contents of 8-Week Old Pigs Fed DSS
#DSS Trials 2014
1	ACACACAC	GTGTGCCAGCMGCCGCGGTAA	GGACTACHVGGGTWTCTAAT	DSS	Base1	DSS_Base1	DSS_29-8_Base1_d7
```

```
#SampleID	BarcodeSequence	LinkerPrimerSequence	ReversePrimer	Trial	Sample	Diet	TrialDiet	SampleTime	TrialSampleTime	Description
#AnalysisoftheBacteriainIntestinalContentsof8-WeekOldPigsFedDSS
#DSSTrials2014
87	ACTGGAGA	GTGTGCCAGCMGCCGCGGTAA	GGACTACHVGGGTWTCTAAT	DSS	Duod	d7	DSS_d7	Duod_d7	DSS_Duod_d7	DSS_29-8_Duod_d7
```

Script:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/validate.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/validate.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Validating Mapping Files"

time validate_mapping_file.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt -o /share/magalab/Kat/DSS/merged.demultiplex/validate.mapping

time validate_mapping_file.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt -o /share/magalab/Kat/DSS/merged.demultiplex/validate.mapping

echo "Check log files"
```

Then, created script to demultiplex (using ```split_libraries.py```) and trim the sequences for specified quality thresholds:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat demultiplex.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Demultiplexing and trimming reads for quality and length."

time split_libraries.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt -f /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.fna -q /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.qual --min_seq_length 200 --max_seq_length 373 --min_qual_score 25 -b 8 -w 25 -z truncate_only -d

time split_libraries.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt -f /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.fna -q /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.qual --min_seq_length 200 --max_seq_length 373 --min_qual_score 25 -b 8 -w 25 -z truncate_only -d

echo "Check output and error"
```

### July 10 2017

After messing with the memory allotted per CPU in SBATCH (by modifying #SBATCH --mem-per-cpu=<MB> in header to reflect 20000 MB), was able to get the scripts to run.

After adjusting the sliding window parameter (-w) to 30 instead of 25, I found that this decreased the number of returned filtered reads, rather than increasing them. I adjusted -w to 20, --min_seq_length and --max_seq_length to try and improve the efficiency a bit.

Results with -w 30:
```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_library_contents$ cat split_library_log.txt
Number raw input seqs	16088672

Length outside bounds of 200 and 373	514840
Num ambiguous bases exceeds limit of 6	0
Missing Qual Score	0
Mean qual score below minimum of 25	7458
Max homopolymer run exceeds limit of 6	54674
Num mismatches in primer exceeds limit of 0: 9997134

Number of sequences with identifiable barcode but without identifiable reverse primer: 70692

-z truncate_only option enabled; sequences without a discernible reverse primer as well as sequences with a valid barcode not found in the mapping file may still be written.

Size of quality score window, in base pairs: 30
Number of sequences where a low quality score window was detected: 22505
Sequences with low quality score window were truncated to the first base of the window.
Sequences discarded after truncation due to sequence length below the minimum 200: 6065

Sequence length details for all sequences passing quality filters:
Raw len min/max/avg	251.0/371.0/301.1
Wrote len min/max/avg	171.0/341.0/253.5

Barcodes corrected/not	11485/4926220
Uncorrected barcodes will not be written to the output fasta file.
Corrected barcodes will be written with the appropriate barcode category.
Corrected but unassigned sequences will not be written unless --retain_unassigned_reads is enabled.

Total valid barcodes that are not in mapping file	0
Sequences associated with valid barcodes that are not in the mapping file will not be written.

Barcodes in mapping file
Num Samples	37
Sample ct min/max/mean: 33 / 41633 / 15737.32
Sample	Sequence Count	Barcode
98	41633	AGAGCTCT
251	33446	CAGATGTC
76	33080	ACGTACGT
124	26018	AGCTAGGA
125	25365	AGTCACAC
115	24203	AGCACATC
116	21061	AGCTAGCT
143	20833	ATCGAAGG
105	19326	AGACGTCT
102	18693	AGTCGAGA
122	18206	AGAGGAGA
97	17363	AGACGAGT
113	16314	AGACGTGA
117	15955	AGCTTGGT
123	15580	AGCACTAC
106	14630	AGAGCTGA
108	13608	AGCTACGT
109	13494	AGCTTGCA
88	13313	AGACCTGT
107	13112	AGCACAAG
252	13024	CAGTGTGT
120	12937	ATCGTACG
100	12750	AGCTACCA
126	12359	AGTCTCAG
114	12299	AGAGGACT
87	12178	ACTGGAGA
110	11645	AGTCGTCA
111	11215	AGTGGAGT
255	11149	CGATCCAA
121	10252	AGACTCAC
254	10146	CATGCATG
253	8529	CATCGAAC
104	8124	ATCGGCAT
127	6898	AGTGTGAG
99	6835	AGCAAGGT
119	6675	AGTGTCAC
142	33	AGTCTGAC

Total number seqs written	582281
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_library_feces$ cat split_library_log.txt
Number raw input seqs	16088672

Length outside bounds of 200 and 373	514840
Num ambiguous bases exceeds limit of 6	0
Missing Qual Score	0
Mean qual score below minimum of 25	7458
Max homopolymer run exceeds limit of 6	54674
Num mismatches in primer exceeds limit of 0: 9997134

Number of sequences with identifiable barcode but without identifiable reverse primer: 101438

-z truncate_only option enabled; sequences without a discernible reverse primer as well as sequences with a valid barcode not found in the mapping file may still be written.

Size of quality score window, in base pairs: 30
Number of sequences where a low quality score window was detected: 37786
Sequences with low quality score window were truncated to the first base of the window.
Sequences discarded after truncation due to sequence length below the minimum 200: 9728

Sequence length details for all sequences passing quality filters:
Raw len min/max/avg	251.0/371.0/301.4
Wrote len min/max/avg	3.0/339.0/253.4

Barcodes corrected/not	22228/4560522
Uncorrected barcodes will not be written to the output fasta file.
Corrected barcodes will be written with the appropriate barcode category.
Corrected but unassigned sequences will not be written unless --retain_unassigned_reads is enabled.

Total valid barcodes that are not in mapping file	0
Sequences associated with valid barcodes that are not in the mapping file will not be written.

Barcodes in mapping file
Num Samples	93
Sample ct min/max/mean: 1835 / 29984 / 10153.94
Sample	Sequence Count	Barcode
131	29984	AGCACTTG
405	29295	GAGATCTC
130	25224	AGAGGTCA
49	24788	ACACCTCT
56	21519	AGACAGAC
27	20475	ACAGTGAG
133	19337	AGTCACTG
128	16593	ATCGTTCC
53	16327	ACGTGTAC
50	16183	ACAGCACT
132	15864	AGCTCAAC
63	15381	ACTGCTCT
47	14841	ACTGAGTC
57	14092	ACACCTGA
39	13920	ACAGACTC
41	13211	ACACCAGT
48	12897	AGACACTC
20	12809	ACGAGATG
29	12776	ACGTCTTC
134	12532	AGTCTCTC
25	12019	ACACAGTC
33	11892	AGACACAG
398	11859	GAGTGTCT
55	11716	ACTGCACA
72	11638	AGACCAGA
42	11466	ACAGAGAC
54	11274	ACTCGACA
135	10790	AGTGTGTC
38	10783	ACAGTGTC
388	10340	GACTTGAC
235	10184	CAGATCTG
51	9994	ACGAACGA
69	9968	ACGTTCGA
37	9822	ACGAGTTC
58	9674	ACAGCAGA
243	9622	CAGATGAG
82	9587	ACAGGACA
8	9400	ACTGGTGT
36	9359	ACGTGAAG
34	9218	ACTGAGAG
62	9070	ACTCGAGT
60	9069	ACGATGGA
59	8928	ACGAAGCA
247	8855	CGATATGC
30	8817	ACTCCAGA
129	8766	AGACTCTG
61	8706	ACGTTCCT
2	8586	ACACTCAG
40	8522	ACACCACA
86	8490	ACTCTCAC
70	8384	ACTCGTCT
71	8343	ACTGCTGA
83	8327	ACGACATC
197	8307	CATCACGA
6	8255	ACTCACTC
16	8218	ACTGTCAG
22	8211	ACTCCACT
218	8177	CACTTGAG
81	8073	ACACGTCA
65	8070	ACACGACT
45	7958	ACGTGATC
24	7947	ACTGTCTC
18	7940	ACACTGAC
66	7903	ACAGCTCA
7	7733	ACTCTGAG
31	7636	ACTGACTG
43	7597	ACGAACCT
3	7526	ACAGGTCT
84	7482	ACGTAGCT
206	7431	CATCTGGA
46	7420	ACTCCTGT
17	7385	ACACAGAG
1	7249	ACACACAC
85	7105	ACGTTGGT
4	7105	ACGACTTG
52	7103	ACGATCGT
67	6950	ACGAAGGT
317	6836	CTAGCTAG
23	6493	ACTGACAC
35	6409	ACTCCTCA
9	6345	ACACACTG
68	5743	ACGTACCA
44	5730	ACGATCCA
14	5406	ACTCAGAC
5	5141	ACGTCAAC
15	5101	ACTCTGTC
19	5042	ACAGTCAC
10	4744	ACACTCTC
11	4316	ACAGGTGA
12	3736	ACGAGAAC
21	3021	ACGTCTAG
248	2121	CGCCTTAT
13	1835	ACGTCATG

Total number seqs written	944316
```

New scripts:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat demultiplex.feces.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.feces.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.feces.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=20000

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Demultiplexing and trimming reads for quality and length."

time split_libraries.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt -f /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.fna -q /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.qual -o /share/magalab/Kat/DSS/merged.demultiplex/split_library_feces --min_seq_length 180 --max_seq_length 380 --min_qual_score 25 --max_primer_mismatch 1 -b 8 -w 20 -z truncate_only --reverse_primer_mismatches 1 -d

echo "Check error and output"
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat demultiplex.contents.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.contents.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/demultiplex.contents.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=20000

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Demultiplexing and trimming reads for quality and length."

time split_libraries.py -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt -f /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.fna -q /share/magalab/Kat/DSS/merged.fastaqual/dss.extendedFrags.qual -o /share/magalab/Kat/DSS/merged.demultiplex/split_library_contents --min_seq_length 180 --max_seq_length 380 --min_qual_score 25 --max_primer_mismatch 1 -b 8 -w 20 -z truncate_only --reverse_primer_mismatches 1 -d

echo "Check output and error"
```

### July 11, 2017

New Logs:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_library_feces$ cat split_library_log.txt
Number raw input seqs	16088672

Length outside bounds of 180 and 380	448135
Num ambiguous bases exceeds limit of 6	0
Missing Qual Score	0
Mean qual score below minimum of 25	7501
Max homopolymer run exceeds limit of 6	58742
Num mismatches in primer exceeds limit of 1: 9917177

Number of sequences with identifiable barcode but without identifiable reverse primer: 56284

-z truncate_only option enabled; sequences without a discernible reverse primer as well as sequences with a valid barcode not found in the mapping file may still be written.

Size of quality score window, in base pairs: 20
Number of sequences where a low quality score window was detected: 55304
Sequences with low quality score window were truncated to the first base of the window.
Sequences discarded after truncation due to sequence length below the minimum 180: 14805

Sequence length details for all sequences passing quality filters:
Raw len min/max/avg	251.0/379.0/301.4
Wrote len min/max/avg	3.0/349.0/252.4

Barcodes corrected/not	23015/4676895
Uncorrected barcodes will not be written to the output fasta file.
Corrected barcodes will be written with the appropriate barcode category.
Corrected but unassigned sequences will not be written unless --retain_unassigned_reads is enabled.

Total valid barcodes that are not in mapping file	0
Sequences associated with valid barcodes that are not in the mapping file will not be written.

Barcodes in mapping file
Num Samples	93
Sample ct min/max/mean: 1872 / 30482 / 10380.83
Sample	Sequence Count	Barcode
131	30482	AGCACTTG
405	29836	GAGATCTC
130	25719	AGAGGTCA
49	25189	ACACCTCT
56	21992	AGACAGAC
27	20945	ACAGTGAG
133	19663	AGTCACTG
128	16898	ATCGTTCC
53	16680	ACGTGTAC
50	16516	ACAGCACT
132	16152	AGCTCAAC
63	15677	ACTGCTCT
47	15090	ACTGAGTC
57	14402	ACACCTGA
39	14241	ACAGACTC
20	13482	ACGAGATG
41	13430	ACACCAGT
48	13135	AGACACTC
29	13080	ACGTCTTC
134	12789	AGTCTCTC
25	12257	ACACAGTC
398	12144	GAGTGTCT
33	12109	AGACACAG
55	11928	ACTGCACA
72	11855	AGACCAGA
42	11709	ACAGAGAC
54	11525	ACTCGACA
38	11021	ACAGTGTC
135	10976	AGTGTGTC
388	10527	GACTTGAC
235	10402	CAGATCTG
51	10179	ACGAACGA
69	10163	ACGTTCGA
37	10162	ACGAGTTC
58	9882	ACAGCAGA
243	9841	CAGATGAG
82	9779	ACAGGACA
8	9620	ACTGGTGT
36	9604	ACGTGAAG
34	9385	ACTGAGAG
60	9301	ACGATGGA
62	9275	ACTCGAGT
59	9088	ACGAAGCA
247	9022	CGATATGC
30	9000	ACTCCAGA
61	8921	ACGTTCCT
129	8920	AGACTCTG
2	8743	ACACTCAG
40	8729	ACACCACA
86	8634	ACTCTCAC
70	8570	ACTCGTCT
83	8522	ACGACATC
71	8507	ACTGCTGA
197	8499	CATCACGA
6	8428	ACTCACTC
16	8414	ACTGTCAG
22	8350	ACTCCACT
218	8345	CACTTGAG
65	8279	ACACGACT
4	8220	ACGACTTG
81	8202	ACACGTCA
45	8140	ACGTGATC
24	8114	ACTGTCTC
66	8079	ACAGCTCA
18	8075	ACACTGAC
7	7931	ACTCTGAG
31	7776	ACTGACTG
43	7746	ACGAACCT
3	7685	ACAGGTCT
84	7631	ACGTAGCT
206	7600	CATCTGGA
46	7597	ACTCCTGT
17	7542	ACACAGAG
1	7502	ACACACAC
52	7280	ACGATCGT
85	7261	ACGTTGGT
67	7108	ACGAAGGT
317	7011	CTAGCTAG
23	6634	ACTGACAC
35	6571	ACTCCTCA
9	6466	ACACACTG
68	5878	ACGTACCA
44	5874	ACGATCCA
14	5526	ACTCAGAC
5	5275	ACGTCAAC
15	5202	ACTCTGTC
19	5160	ACAGTCAC
10	4860	ACACTCTC
11	4402	ACAGGTGA
12	3939	ACGAGAAC
21	3090	ACGTCTAG
248	2155	CGCCTTAT
13	1872	ACGTCATG

Total number seqs written	965417
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_library_contents$ cat split_library_log.txt
Number raw input seqs	16088672

Length outside bounds of 180 and 380	448135
Num ambiguous bases exceeds limit of 6	0
Missing Qual Score	0
Mean qual score below minimum of 25	7501
Max homopolymer run exceeds limit of 6	58742
Num mismatches in primer exceeds limit of 1: 9917177

Number of sequences with identifiable barcode but without identifiable reverse primer: 43342

-z truncate_only option enabled; sequences without a discernible reverse primer as well as sequences with a valid barcode not found in the mapping file may still be written.

Size of quality score window, in base pairs: 20
Number of sequences where a low quality score window was detected: 33137
Sequences with low quality score window were truncated to the first base of the window.
Sequences discarded after truncation due to sequence length below the minimum 180: 9336

Sequence length details for all sequences passing quality filters:
Raw len min/max/avg	251.0/379.0/301.1
Wrote len min/max/avg	151.0/347.0/252.5

Barcodes corrected/not	11886/5054128
Uncorrected barcodes will not be written to the output fasta file.
Corrected barcodes will be written with the appropriate barcode category.
Corrected but unassigned sequences will not be written unless --retain_unassigned_reads is enabled.

Total valid barcodes that are not in mapping file	0
Sequences associated with valid barcodes that are not in the mapping file will not be written.

Barcodes in mapping file
Num Samples	37
Sample ct min/max/mean: 30 / 42247 / 16044.68
Sample	Sequence Count	Barcode
98	42247	AGAGCTCT
251	34112	CAGATGTC
76	33881	ACGTACGT
124	26509	AGCTAGGA
125	25856	AGTCACAC
115	24663	AGCACATC
116	21453	AGCTAGCT
143	21186	ATCGAAGG
105	19612	AGACGTCT
102	18938	AGTCGAGA
122	18517	AGAGGAGA
97	17685	AGACGAGT
113	16749	AGACGTGA
117	16241	AGCTTGGT
123	15852	AGCACTAC
106	14930	AGAGCTGA
108	13868	AGCTACGT
109	13754	AGCTTGCA
88	13462	AGACCTGT
107	13339	AGCACAAG
252	13317	CAGTGTGT
120	13160	ATCGTACG
100	13043	AGCTACCA
126	12586	AGTCTCAG
114	12569	AGAGGACT
87	12326	ACTGGAGA
110	11846	AGTCGTCA
111	11484	AGTGGAGT
255	11362	CGATCCAA
121	10441	AGACTCAC
254	10347	CATGCATG
253	8710	CATCGAAC
104	8698	ATCGGCAT
127	7037	AGTGTGAG
99	7021	AGCAAGGT
119	6822	AGTGTCAC
142	30	AGTCTGAC

Total number seqs written	593653
```

### July 16, 2017

Trying new methods for demultiplexing to improve efficiency, as most reads are not demultiplexed due to errors in the primer or barcode location/sequence.

Using additional parameters in ```split_libraries_fastq.py``` in QIIME to allow for conversion between FASTA and FASTQ in the future.

Tried cutadapt to remove any contaminating Illumina primer sequence, rather than trimmomatic. According to documents from Illumina and other bioinformatics sites, the following sequences are fairly ubiquitous in Illumina's adapters and may serve as a potential source of contamination:

Truseq Universal Adapter:
AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT

Multiplexing Read 1 Sequencing Primer:
ACACTCTTTCCCTACACGACGCTCTTCCGATCT

Multiplexing Index Read Sequencing Primer:
GATCGGAAGAGCACACGTCTGAACTCCAGTCAC

Multiplexing Read 2 Sequencing Primer:
GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT

(Sources: https://support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/experiment-design/illumina-adapter-sequences_1000000002694-01.pdf, http://bioinformatics.cvr.ac.uk/blog/illumina-adapter-and-primer-sequences/)

Used cutadapt to remove any Universal Adapter sequence from both the 5' and 3' ends of the reads, allowing for 0.5% error in adapter sequence matching. Then re-extracted barcodes and used ```split_libraries_fastq.py``` to demultiplex. If this does not work, I will attempt to use cutadapt to remove all of the above sequences, and then, I'll try using trimmomatic again.

Scripts for cutadapt:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat ../cutadapt/cutadapt5p.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/cutadapt/5p.out
#SBATCH -e /share/magalab/Kat/DSS/cutadapt/5p.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Loading cutadapt."

module load cutadapt/1.8

echo "Running cutadapt on merged reads to remove potentially degraded, universal Illumina TruSeq adapter from the 5' end."

cutadapt -g AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed5.fastq /share/magalab/Kat/DSS/merged.reads/dss.extendedFrags.fastq

echo "Check .out and .err"
```

Log:

```
=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                    92,275 (0.6%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,893,533,949 bp
Total written (filtered):  4,893,097,188 bp (100.0%)

=== Adapter 1 ===

Sequence: AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT; Type: regular 5'; Length: 58; Trimmed: 92275 times.

No. of allowed errors:
0-19 bp: 0; 20-39 bp: 1; 40-58 bp: 2
```

Second Script:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/cutadapt/3p.out
#SBATCH -e /share/magalab/Kat/DSS/cutadapt/3p.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Loading cutadapt."

module load cutadapt/1.8

echo "Running cutadapt on merged reads to remove potentially degraded, universal Illumina TruSeq adapter from the 5' end."

cutadapt -a AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed5.fastq

echo "Check .out and .err"
```

Log:

```
=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                    37,631 (0.2%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,893,097,188 bp
Total written (filtered):  4,892,972,043 bp (100.0%)

=== Adapter 1 ===

Sequence: AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT; Type: regular 3'; Length: 58; Trimmed: 37631 times.

No. of allowed errors:
0-19 bp: 0; 20-39 bp: 1; 40-58 bp: 2

Bases preceding removed adapters:
  A: 31.7%
  C: 24.8%
  G: 22.2%
  T: 21.3%
  none/other: 0.0%
```

### July 17, 2017

Demultiplexed samples again using ```split_libraries_fastq.py```

Scripts:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat split.lib.fastq.cut.contents.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/contents.cut.fastq.demultiplex.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/contents.cut.fastq.demultiplex.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=10000

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Running extract_barcodes.py in order to use split_libraries_fastq.py."

time extract_barcodes.py -f /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq -o /share/magalab/Kat/DSS/merged.demultiplex/cut_contents_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt -a

echo "Demultiplexing with split_libraries_fastq.py."

time split_libraries_fastq.py -i /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq -o /share/magalab/Kat/DSS/merged.demultiplex/split_lib_cut_fastq_out_contents/ -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt -b /share/magalab/Kat/DSS/merged.demultiplex/cut_contents_barcodes/barcodes.fastq --store_qual_scores --store_demultiplexed_fastq -q 24 --barcode_type 8 --phred_offset 33

echo "Check .err and .out files"
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat split.lib.fastq.cut.feces.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/merged.demultiplex/feces.cut.fastq.demultiplex.out
#SBATCH -e /share/magalab/Kat/DSS/merged.demultiplex/feces.cut.fastq.demultiplex.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=10000

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Running extract_barcodes.py in order to use split_libraries_fastq.py."

time extract_barcodes.py -f /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq -o /share/magalab/Kat/DSS/merged.demultiplex/cut_feces_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt -a

echo "Demultiplexing with split_libraries_fastq.py."

time split_libraries_fastq.py -i /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq -o /share/magalab/Kat/DSS/merged.demultiplex/split_lib_cut_fastq_out_feces/ -m /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt -b /share/magalab/Kat/DSS/merged.demultiplex/cut_feces_barcodes/barcodes.fastq --store_qual_scores --store_demultiplexed_fastq -q 24 --barcode_type 8 --phred_offset 33

echo "Check .err and .out files"
```

Logs:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_lib_cut_fastq_out_feces$ cat split_library_log.txt
Input file paths
Mapping filepath: /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_feces_w.rev.txt (md5: 02213fbb49810b4840990163b588b2d4)
Sequence read filepath: /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq (md5: 460db2ea58c71dae5f59338697f4af88)
Barcode read filepath: /share/magalab/Kat/DSS/merged.demultiplex/cut_feces_barcodes/barcodes.fastq (md5: 6ad37428fdb22d7f22d87ce7e7c61795)

Quality filter results
Total number of input sequences: 16088672
Barcode not in mapping file: 15097530
Read too short after quality truncation: 23795
Count of N characters exceeds limit: 0
Illumina quality digit = 0: 0
Barcode errors exceed max: 0

Result summary (after quality filtering)
Median sequence length: 302.00
131	30546
405	29480
130	26030
49	24823
56	21905
27	21152
133	19984
128	16941
50	16538
53	16537
132	16314
63	15709
47	15058
57	14308
39	14258
20	13452
41	13367
134	13143
48	13051
29	12937
33	12264
25	12252
55	11961
72	11807
42	11804
398	11647
135	11524
54	11404
38	10948
8	10803
235	10279
51	10209
388	10172
69	10114
37	10078
58	9910
82	9850
243	9691
129	9493
36	9471
62	9465
34	9428
60	9289
59	9138
247	9033
30	8892
2	8874
61	8842
86	8624
40	8611
16	8526
218	8520
70	8487
197	8475
71	8444
83	8435
6	8381
22	8280
4	8220
81	8200
46	8190
65	8150
18	8069
7	8068
45	8048
66	8041
24	8027
31	7995
43	7760
17	7729
3	7657
84	7639
206	7441
52	7351
1	7340
317	7329
85	7313
67	7224
23	6634
9	6619
35	6426
68	5845
44	5810
14	5482
5	5212
15	5166
19	5081
10	4899
11	4500
12	3928
21	3074
248	2168
13	1754

Total number seqs written	967347
---
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex/split_lib_cut_fastq_out_contents$ cat split_library_log.txt
Input file paths
Mapping filepath: /share/magalab/Kat/DSS/merged.demultiplex/mapping.files/MappingFiles_DSS_contents_w.rev.txt (md5: 7f23df669ac86052c0eb1cf5feebcecb)
Sequence read filepath: /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq (md5: 460db2ea58c71dae5f59338697f4af88)
Barcode read filepath: /share/magalab/Kat/DSS/merged.demultiplex/cut_contents_barcodes/barcodes.fastq (md5: 6ad37428fdb22d7f22d87ce7e7c61795)

Quality filter results
Total number of input sequences: 16088672
Barcode not in mapping file: 15468968
Read too short after quality truncation: 14569
Count of N characters exceeds limit: 0
Illumina quality digit = 0: 0
Barcode errors exceed max: 0

Result summary (after quality filtering)
Median sequence length: 302.00
98	42270
76	33898
251	33726
124	27233
125	25897
115	25305
143	22025
116	21892
105	19730
102	19491
122	19042
97	18284
113	17067
117	16785
123	16012
106	15195
108	14377
109	14049
252	14037
107	13788
88	13607
120	13552
100	13413
126	13020
114	12965
87	12136
111	12105
110	12050
255	11339
121	10735
254	10327
104	8905
253	8550
99	7489
127	7477
119	7358
142	4

Total number seqs written	605135
---
```

Now running cutadapt again to try and remove any additional adapter/primer sequences (see above).

Script:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/merged.demultiplex$ cat ../cutadapt/cutadapt.long.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/cutadapt/long.out
#SBATCH -e /share/magalab/Kat/DSS/cutadapt/long.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=10000

echo "Loading cutadapt."

module load cutadapt/1.8

echo "Running cutadapt on merged reads to remove potentially degraded, Multiplexing Read 1 Sequencing adapter from the 5' end."

cutadapt -g ACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.5p.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq

echo "Running cutadapt to remove potentially degraded, Multiplexing Read 1 Sequencing adapter from the 3' end."

cutadapt -a ACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.5p.fastq

echo "Running cutadapt to remove potentially degraded, Multiplexing index Read sequencing primer from the 5' end."

cutadapt -g GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.5p.fastq /share/magalab/Kat/DSS/cutagapt/dss.merged.trimmed.mplex1.fastq

echo "Running cutadapt to remove potentially degraded, Multiplexing index read sequencing primer from the 3' end."

cutadapt -a GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.5p.fastq

echo "Running cutadapt to remove potentially degraded, Multiplexing Read 2 Sequencing adapter from the 5' end."

cutadapt -g GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.5p.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.fastq

echo "Running cutadapt to remove potenitally degraded, Multiplexing Read 2 Sequencing adapter from the 3' end."

cutadapt -g GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.5p.fastq

echo "Check .out and .err"
```

Lastly, will try demultiplexing again with ```split_libraries_fastq.py``` and ```split_libraries.py``` to see which run is most efficient at correctly retrieving as many reads as possible.

Logs from cutadapt:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/cutadapt$ cat long.out
Loading cutadapt.
Running cutadapt on merged reads to remove potentially degraded, Multiplexing Read 1 Sequencing adapter from the 5' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -g ACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.5p.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 560.37 s (35 us/read; 1.72 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                       492 (0.0%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,892,972,043 bp
Total written (filtered):  4,892,959,695 bp (100.0%)

=== Adapter 1 ===

Sequence: ACACTCTTTCCCTACACGACGCTCTTCCGATCT; Type: regular 5'; Length: 33; Trimmed: 492 times.

No. of allowed errors:
0-19 bp: 0; 20-33 bp: 1

Overview of removed sequences
length	count	expect	max.err	error counts
3	450	251385.5	0	450
4	3	62846.4	0	3
5	1	15711.6	0	1
115	2	0.0	1	0 2
146	1	0.0	1	1
174	1	0.0	1	0 1
279	1	0.0	1	0 1
284	22	0.0	1	21 1
302	1	0.0	1	1
304	1	0.0	1	0 1
311	1	0.0	1	0 1
330	1	0.0	1	0 1
339	1	0.0	1	0 1
359	1	0.0	1	0 1
361	1	0.0	1	0 1
378	1	0.0	1	0 1
396	1	0.0	1	0 1
410	1	0.0	1	0 1
414	1	0.0	1	0 1

Running cutadapt to remove potentially degraded, Multiplexing Read 1 Sequencing adapter from the 3' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -a ACACTCTTTCCCTACACGACGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.5p.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 544.12 s (34 us/read; 1.77 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                   249,957 (1.6%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,892,959,695 bp
Total written (filtered):  4,891,918,267 bp (100.0%)

=== Adapter 1 ===

Sequence: ACACTCTTTCCCTACACGACGCTCTTCCGATCT; Type: regular 3'; Length: 33; Trimmed: 249957 times.

No. of allowed errors:
0-19 bp: 0; 20-33 bp: 1

Bases preceding removed adapters:
  A: 4.7%
  C: 77.5%
  G: 11.9%
  T: 5.9%
  none/other: 0.0%

Overview of removed sequences
length	count	expect	max.err	error counts
3	78698	251385.5	0	78698
4	89919	62846.4	0	89919
5	45029	15711.6	0	45029
6	33808	3927.9	0	33808
7	2448	982.0	0	2448
8	39	245.5	0	39
9	3	61.4	0	3
10	9	15.3	0	9
12	1	1.0	0	1
27	1	0.0	1	1
29	1	0.0	1	0 1
32	1	0.0	1	0 1

Running cutadapt to remove potentially degraded, Multiplexing index Read sequencing primer from the 5' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -g GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.5p.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 591.71 s (37 us/read; 1.63 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                   206,921 (1.3%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,891,918,267 bp
Total written (filtered):  4,891,003,948 bp (100.0%)

=== Adapter 1 ===

Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: regular 5'; Length: 33; Trimmed: 206921 times.

No. of allowed errors:
0-19 bp: 0; 20-33 bp: 1

Overview of removed sequences
length	count	expect	max.err	error counts
3	82124	251385.5	0	82124
4	30566	62846.4	0	30566
5	33696	15711.6	0	33696
6	47649	3927.9	0	47649
7	12872	982.0	0	12872
8	3	245.5	0	3
33	2	0.0	1	2
52	1	0.0	1	1
105	1	0.0	1	1
107	1	0.0	1	1
117	1	0.0	1	1
118	1	0.0	1	1
124	1	0.0	1	1
142	1	0.0	1	1
149	1	0.0	1	1
201	1	0.0	1	0 1

Running cutadapt to remove potentially degraded, Multiplexing index read sequencing primer from the 3' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCAC -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.5p.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 558.77 s (35 us/read; 1.73 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                   428,917 (2.7%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,891,003,948 bp
Total written (filtered):  4,889,443,908 bp (100.0%)

=== Adapter 1 ===

Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: regular 3'; Length: 33; Trimmed: 428917 times.

No. of allowed errors:
0-19 bp: 0; 20-33 bp: 1

Bases preceding removed adapters:
  A: 20.6%
  C: 42.3%
  G: 14.6%
  T: 22.6%
  none/other: 0.0%

Overview of removed sequences
length	count	expect	max.err	error counts
3	169638	251385.5	0	169638
4	245899	62846.4	0	245899
5	13015	15711.6	0	13015
6	212	3927.9	0	212
7	115	982.0	0	115
8	1	245.5	0	1
9	28	61.4	0	28
10	1	15.3	0	1
11	1	3.8	0	1
12	4	1.0	0	4
14	2	0.1	0	2
21	1	0.0	1	0 1

Running cutadapt to remove potentially degraded, Multiplexing Read 2 Sequencing adapter from the 5' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -g GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.5p.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 573.12 s (36 us/read; 1.68 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                     8,251 (0.1%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,889,443,908 bp
Total written (filtered):  4,889,418,210 bp (100.0%)

=== Adapter 1 ===

Sequence: GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT; Type: regular 5'; Length: 34; Trimmed: 8251 times.

No. of allowed errors:
0-19 bp: 0; 20-34 bp: 1

Overview of removed sequences
length	count	expect	max.err	error counts
3	8155	251385.5	0	8155
4	90	62846.4	0	90
33	1	0.0	1	0 1
52	2	0.0	1	0 2
54	1	0.0	1	0 1
328	1	0.0	1	0 1
354	1	0.0	1	0 1

Running cutadapt to remove potenitally degraded, Multiplexing Read 2 Sequencing adapter from the 3' end.
This is cutadapt 1.8 with Python 2.7.6
Command line parameters: -g GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT -e 0.05 -o /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.fastq /share/magalab/Kat/DSS/cutadapt/dss.merged.trimmed.mplex1.index.mplex2.5p.fastq
Trimming 1 adapter(s) with at most 5.0% errors in single-end mode ...
Finished in 569.98 s (35 us/read; 1.69 M reads/minute).

=== Summary ===

Total reads processed:              16,088,672
Reads with adapters:                         1 (0.0%)
Reads written (passing filters):    16,088,672 (100.0%)

Total basepairs processed: 4,889,418,210 bp
Total written (filtered):  4,889,418,207 bp (100.0%)

=== Adapter 1 ===

Sequence: GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT; Type: regular 5'; Length: 34; Trimmed: 1 times.

No. of allowed errors:
0-19 bp: 0; 20-34 bp: 1

Overview of removed sequences
length	count	expect	max.err	error counts
3	1	251385.5	0	1

Check .out and .err
```

### July 21, 2017

Tried rerunning ```split_libraries_fastq.py``` on data trimmed with cutadapt to remove all four possible contaminating adapters. This resulted in even less efficiency with demultiplexing (only ~940,000 reads returned for feces and ~580,000 reads for contents), so I returned to using Trimmomatic in PE mode on the original FASTQ reads, followed by merging and demultiplexing with both ```split_libraries.py``` and ```split_libraries_fastq.py```.

As a last ditch effort, I am also trying to run ```split_libraries.py``` on the original output from cutadapt (where only one potential adapter sequence is checked), to see if this is at all more efficient than ```split_libraries_fastq.py```.

Scripts for Trimmomatic in PE mode:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/trimmomatic/trim.pe.out
#SBATCH -e /share/magalab/Kat/DSS/trimmomatic/trim.pe.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00
#SBATCH --mem-per-cpu=10000

echo "Loading Trimmomatic."

module load trimmomatic/0.33

echo "Running Trimmomatic in Paired End mode, using palindromic clipping to remove short contaminating adapter sequences."

trimmomatic PE -threads 4 /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq ./dss.trimmed.fwd.paired.fastq ./dss.trimmed.fwd.unpaired.fastq ./dss.trimmed.rev.paired.fastq ./dss.trimmed.rev.unpaired.fastq ILLUMINACLIP:adapters.fa:1:30:10:2:keepBothReads TRAILING:25 SLIDINGWINDOW:5:25 MINLEN:50

echo "Check error and out files."
```

Output from trimmomatic:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/trimmomatic$ cat trim.pe.err
Module trimmomatic-0.33-static loaded.
TrimmomaticPE: Started with arguments: -threads 4 /share/magalab/Kat/DSS/DSS_S1_L001_R1_001.fastq /share/magalab/Kat/DSS/DSS_S1_L001_R2_001.fastq ./dss.trimmed.fwd.paired.fastq ./dss.trimmed.fwd.unpaired.fastq ./dss.trimmed.rev.paired.fastq ./dss.trimmed.rev.unpaired.fastq ILLUMINACLIP:adapters.fa:1:30:10:2:keepBothReads TRAILING:25 SLIDINGWINDOW:5:25 MINLEN:50
Using PrefixPair: 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
Using PrefixPair: 'CACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'CACTCTTTCCCTACACGACGCTCTTCCGATCT'
Using PrefixPair: 'GATCGGAAGAGCACACGTCTGAACTCCAGTCAC' and 'GATCGGAAGAGCACACGTCTGAACTCCAGTCAC'
Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
Using PrefixPair: 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT'
Using Long Clipping Sequence: 'AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC'
Using Long Clipping Sequence: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT'
Using Long Clipping Sequence: 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
Using Long Clipping Sequence: 'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA'
ILLUMINACLIP: Using 5 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Quality encoding detected as phred33
Input Read Pairs: 16765099 Both Surviving: 12667652 (75.56%) Forward Only Surviving: 3272018 (19.52%) Reverse Only Surviving: 277449 (1.65%) Dropped: 547980 (3.27%)
TrimmomaticPE: Completed successfully
```

After verifying average length of the reads with fastqc, I then merged the resulting ```dss.trimmed.fwd.paired.fastq``` and ```dss.trimmed.rev.paired.fastq``` files using Flash2:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/trimmed.merged$ cat trimmed.merge.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/trimmed.merged/merge.out
#SBATCH -e /share/magalab/Kat/DSS/trimmed.merged/merge.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading Flash2 module"
module load flash2/c41a82e

echo "Average read length 245 bp (trimmed), average fragment length should be between 290 and 300, and standard deviation of read lengths approximately 29."

echo "Options -r 245, -f 290, and -s 29 are used to calculate max overlap"

time flash2 /share/magalab/Kat/DSS/trimmomatic/dss.trimmed.fwd.paired.fastq /share/magalab/Kat/DSS/trimmomatic/dss.trimmed.rev.paired.fastq -o dss.trimmed -r 245 -f 290 -s 29

echo "Merging complete"
```

Lastly, I concatenated the merged read file with any unpaired forward reads (which should still have a barcode), to ensure that I have the maximum number of usable reads for subsequent steps:

```
cat /share/magalab/Kat/DSS/trimmomatic/dss.trimmed.fwd.unpaired.fastq /share/magalab/Kat/DSS/trimmed.merged/dss.trimmed.extendedFrags.fastq /share/magalab/Kat/DSS/trimmed.merged/dss.trimmed.notCombined_1.fastq > ./dss.trimmed.merged.fwd.fastq
```

Now, I have a new file which may be used for demultiplexing (again). This file has a total of about ~15.9 M reads. Barcodes will hopefully be present at the start of most of these reads, as the Kmer content report in FASTQC showed an overrepresentation of a few kmers at the very start of the reads.
