# DSS/DSS-hLZ Workflow and History (Part 3)

### June 7, 2017

**Working with QIIME**

Tried to validate mapping files by:

Loading QIIME: ```module load qiime/1.9.1```

Running: ```validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -o /share/magalab/Kat/DSS/qiime/feces.mapping_output```

Received following error:

```
(validate_mapping_file.py:30957): Gdk-CRITICAL **: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed
Traceback (most recent call last):
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/bin/validate_mapping_file.py", line 14, in <module>
    from qiime.util import parse_command_line_parameters, get_options_lookup,\
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/qiime/util.py", line 49, in <module>
    from skbio.util import remove_files, create_dir
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/__init__.py", line 15, in <module>
    import skbio.io
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/io/__init__.py", line 309, in <module>
    import_module('skbio.io.clustal')
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/io/clustal.py", line 123, in <module>
    from skbio.alignment import Alignment
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/alignment/__init__.py", line 230, in <module>
    from ._alignment import Alignment, SequenceCollection, StockholmAlignment
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/alignment/_alignment.py", line 21, in <module>
    from skbio.stats.distance import DistanceMatrix
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/stats/distance/__init__.py", line 193, in <module>
    from ._base import (DissimilarityMatrixError, DistanceMatrixError,
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/skbio/stats/distance/_base.py", line 16, in <module>
    import matplotlib.pyplot as plt
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.py", line 109, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/backend_gtk3agg.py", line 11, in <module>
    from . import backend_gtk3
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/matplotlib/backends/backend_gtk3.py", line 55, in <module>
    cursors.MOVE          : Gdk.Cursor.new(Gdk.CursorType.FLEUR),
TypeError: constructor returned NULL
```

### June 8, 2017

After contacting Genome Center, error was due to QIIME attempting to run matplotlib in sbatch. Needed to change the ```matplotlibrc``` file, specifically altering the backend option, to 'Agg'.

To do this, visited:
http://matplotlib.org/users/customizing.html

Scrolled to bottom of page and copied link to sample matplotlibrc file. In terminal, within the directory where I wanted to download the file, used ```wget``` to download file from the webpage. Then, altered ONLY the backend option to 'Agg' using ```nano```.

As default, QIIME will look for ```matplotlibrc``` file in the closest possible place. If running from sbatch files from a particular directory, place the edited ```matplotlibrc``` file there, and QIIME will look there first and find the appropriate, edited file.

Portion of ```matplotlibrc``` changed:

```
#### CONFIGURATION BEGINS HERE

# The default backend; one of GTK GTKAgg GTKCairo GTK3Agg GTK3Cairo
# MacOSX Qt4Agg Qt5Agg TkAgg WX WXAgg Agg Cairo GDK PS PDF SVG
# Template.
# You can also deploy your own backend outside of matplotlib by
# referring to the module name (which must be in the PYTHONPATH) as
# 'module://my_backend'.
backend      : Agg

```

QIIME now functions normally.

### June 9, 2017

After copying and pasting the older version of the DSS contents mapping file into a new Excel spreadsheet and saving as a Tab delimited text (.txt) file, was able to validate mapping files with no issues.

Script for validating the mapping files for both feces and contents sample sets:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/validate.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/validate.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Validating Mapping Files"

time validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -o /share/magalab/Kat/DSS/qiime/validate.mapping

time validate_mapping_file.py -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -o /share/magalab/Kat/DSS/qiime/validate.mapping

echo "Check log files"
```

Next step is to demultiplex. Scripts for demultiplexing for both feces and contents sample sets:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/demultiplex.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/demultiplex.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Running extract_barcodes.py in order to use split_libraries_fastq.py later on."

time extract_barcodes.py -f /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.feces_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/qiime/mapping.file/MappingFiles_DSS_feces.txt -a

time extract_barcodes.py -f /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.contents_barcodes -c barcode_single_end -l 8 -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -a

echo "Running split_libraries_fastq.py to demultiplex."

time split_libraries_fastq.py  -i /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.feces_split_library_output -b /share/magalab/Kat/DSS/qiime/dss.feces_barcodes/barcodes.fastq -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_feces.txt -q 34 --phred_offset 33

time split_libraries_fastq.py  -i /share/magalab/Kat/DSS/qiime/dss.merged.trimmed.screened.good.fastq -o /share/magalab/Kat/DSS/qiime/dss.contents_split_library_output -b /share/magalab/Kat/DSS/qiime/dss.contents_barcodes/barcodes.fastq -m /share/magalab/Kat/DSS/qiime/mapping.files/MappingFiles_DSS_contents.txt -q 34 --phred_offset 33

echo "Demultiplexed files ready for Chimera detection."
```

Notes about options:

For ```extract_barcodes.py```

 * **-c barcode_single_end** indicates that input was a single fastq file, with the barcode located at the beginning of each read.
 * **-l 8** indicates that barcodes are 8 bp long.
 * **-a** (--attempt_read_reorientation) "Will attempt to search for the forward and reverse primer in the read and adjust the sequence orientation to match the orientation of the forward primer. An exact match for the forward and reverse complemented versions of the primers are tested for, and sequences are reverse complemented, if necessary, before writing. Sequences without an exact match are written to a separate output fastq file, labeled as _no_primer_match.fastq"

For ```split_libraries_fastq.py```

 * **-b filepath** indicates path to barcodes.fastq file, generated from ```extract_barcodes.py```
 * **-q 34** indicates maximum UNACCEPTABLE Phred quality score. QIIME will only accept reads with Q 35 or higher. Value arbitrarily decided based on FASTQC output for ```dss.merged.trimmed.screened.good.fastq```.
 * **--phred_offset 33** specifies the ascii offset for decoding phred scores.


### June 19, 2017

Checked that files were demultiplexed correctly, received error:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/qiime$ cat demultiplex.err
Module qiime-1.9.1-x86_64-linux-ubuntu14.04 loaded.

real	20m7.417s
user	15m1.368s
sys	0m21.812s

real	19m18.049s
user	14m45.636s
sys	0m19.508s
Error in split_libraries_fastq.py: Some or all barcodes are not valid golay codes. Do they need to be reverse complemented? If these are not golay barcodes pass --barcode_type 12 to disable barcode error correction, or pass --barcode_type # if the barcodes are not 12 base pairs, where # is the size of the barcodes. Invalid codes:
	ACTCAGAC CACTTGAG ACTGACTG ACAGCACT AGCACTTG ACGAACGA ACGATCCA ACAGGACA ACACGACT ACTCGTCT ACAGAGAC AGAGGTCA AGTGTGTC ACGTCATG CGCCTTAT ACACCAGT ACTGCTGA AGTCTCTC ACGAGAAC GACTTGAC CAGATCTG ACACTGAC ACAGTGTC ACTCGACA ACAGCTCA ACTGAGTC ACACTCAG ACGTGAAG ACACAGTC CATCACGA GAGTGTCT ACAGTGAG ACGTCAAC ACGAGATG ACGTAGCT AGACCAGA ACGACATC ACGTACCA ACGTTCCT ACTCTCAC ACTGGTGT ACGTCTAG ACGAGTTC ATCGTTCC ACTGTCAG AGACAGAC ACTCCTCA AGACACAG ACAGTCAC AGACTCTG CAGATGAG ACACCTGA ACAGCAGA ACACGTCA ACTGAGAG ACGAACCT ACAGACTC ACGATGGA ACGTCTTC AGCTCAAC ACTGTCTC ACGATCGT ACTCGAGT ACAGGTCT ACTCTGAG ACTGCTCT ACACCACA ACGAAGGT ACTGACAC ACACTCTC AGTCACTG ACGACTTG ACTCCAGA ACAGGTGA ACACACAC ACTCACTC ACTGCACA ACGTTGGT GAGATCTC ACACAGAG CGATATGC ACGAAGCA ACACACTG ACGTTCGA CTAGCTAG AGACACTC CATCTGGA ACTCCACT ACGTGTAC ACGTGATC ACACCTCT ACTCTGTC ACTCCTGT

If you need help with QIIME, see:
http://help.qiime.org

real	0m2.557s
user	0m1.584s
sys	0m2.356s
Error in split_libraries_fastq.py: Some or all barcodes are not valid golay codes. Do they need to be reverse complemented? If these are not golay barcodes pass --barcode_type 12 to disable barcode error correction, or pass --barcode_type # if the barcodes are not 12 base pairs, where # is the size of the barcodes. Invalid codes:
	AGAGCTCT ACTGGAGA ACGTACGT ATCGTACG AGCAAGGT CAGATGTC CAGTGTGT AGACTCAC AGCACTAC AGTCTGAC AGTCGAGA AGAGCTGA AGTCGTCA AGTGTCAC AGCTAGCT AGCTTGGT AGTCTCAG AGCACATC AGCACAAG AGTGTGAG AGAGGACT CATCGAAC AGACGAGT AGCTACGT AGCTAGGA AGCTTGCA AGACCTGT AGCTACCA AGAGGAGA ATCGGCAT AGACGTGA CGATCCAA AGACGTCT ATCGAAGG AGTGGAGT AGTCACAC CATGCATG

If you need help with QIIME, see:
http://help.qiime.org

real	0m1.686s
user	0m1.568s
sys	0m2.564s
```

Adjusted ```dss.demultiplex.sh``` and scripts for ```split_libraries_fastq.py``` to include option ```--barcode_type 8```.

### June 28, 2017

Both feces and content samples were correctly demultiplexed using the previous adjustments.

Having issues running usearch application in QIIME.

First, created a test file using the following command:

```  347  sed -n -e '1,1000p' dss.feces_split_library_output/seqs.fna > dss.test.fasta```

Then, attempted to test different parameters on test file for ```identify_chimeric_seqs.py```

Attempted script:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/qiime$ cat test.id.chimera.sh
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/test.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/test.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Testing default usearch61 parameters on test file."

time identify_chimeric_seqs.py -i /share/magalab/Kat/DSS/qiime/dss.test.fasta -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set_aligned/97_otus.fasta -m usearch61 --usearch61_minh 0.28 --usearch61_xn 8.0 -o /share/magalab/Kat/DSS/qiime/test_default_id.chimera_output

echo "Testing new parameters on test file."

time identify_chimeric_seqs.py -i /share/magalab/Kat/DSS/qiime/dss.test.fasta -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set_aligned/97_otus.fasta -m usearch61 --usearch61_minh 1.28 --usearch61_xn 5.0 -o /share/magalab/Kat/DSS/test_new.xn.minh_id.chimeras_output

echo "Testing new mindiffs parameter on test file."

time identify_chimeric_seqs.py -i /share/magalab/Kat/DSS/qiime/dss.test.fasta -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set_aligned/97_otus.fasta -m usearch61 --usearch61_minh 1.28 --usearch61_xn 5.0 --usearch61_mindiffs 5 -o /share/magalab/Kat/DSS/qiime/test_new.mindiffs_id.chimera_output

echo "Finished testing."
```

Received the same error for all three QIIME commands in the script:


```
Traceback (most recent call last):
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/bin/identify_chimeric_seqs.py", line 354, in <module>
    main()
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/bin/identify_chimeric_seqs.py", line 350, in main
    threads=threads)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/qiime/identify_chimeric_seqs.py", line 774, in usearch61_chimera_check
    log_lines, verbose, threads)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/qiime/identify_chimeric_seqs.py", line 891, in identify_chimeras_usearch61
    consout_filepath=output_consensus_fp)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/bfillings/usearch.py", line 2278, in usearch61_smallmem_cluster
    app = Usearch61(params, WorkingDir=output_dir, HALT_EXEC=HALT_EXEC)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 201, in __init__
    self._error_on_missing_application(params)
  File "/software/qiime/1.9.1/x86_64-linux-ubuntu14.04/usr/local/lib/python2.7/dist-packages/burrito/util.py", line 468, in _error_on_missing_application
    "Is it in your path?" % command)
burrito.util.ApplicationNotFoundError: Cannot find usearch61. Is it installed? Is it in your path?
```

Attempted several times to download, install, and change the path to usearch (v. 6.1, which is compatible with QIIME), as described on QIIME's website. First in the qiime directory, then in a newly made directory called bin.

After downloading usearch, I moved the binary into both directories on the server using ```scp -r```

Then, I renamed the file as directed on QIIME's website (http://qiime.org/install/alternative.html#native-base), using the command:

```  447  ln -s /share/magalab/Kat/bin/usearch6.1.544_i86linux32 /share/magalab/Kat/bin/usearch```

Lastly, I attempted to change the path variable to include the path to the usearch binary.

```
488  "export PATH=/share/magalab/Kat/DSS/qiime/:$PATH" >> $HOME/.bashrc
490  echo $PATH

493  echo "export PATH=/share/magalab/Kat/bin/:$PATH" >> $HOME/.bashrc
494  source $HOME/.bashrc
```

Checked the $PATH variable using ```echo $PATH``` and it appeared correct, but scripts still will not work.

Contacted UC Davis Genome Center for help.

### June 30, 2017

After Genome Center added usearch to their path, I was able to check for chimeras and use ```filter_fasta.py``` to remove chimeric sequences.

After first running the test file, I determined that there were no differences from the use of difference chimera detection parameters, so I ran default parameters for my file.

Final script: ```dss.id.chimera.sh```

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/chimeras.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/chimeras.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Checking for chimeras in feces and contents samples using usearch61 method with default parameters"

time identify_chimeric_seqs.py -i /share/magalab/Kat/DSS/qiime/dss.feces_split_library_output/seqs.fna -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set/97_otus.fasta -m usearch61 --usearch61_minh 0.28 --usearch61_xn 8.0 -o /share/magalab/Kat/DSS/qiime/dss.feces_chimera_output

time identify_chimeric_seqs.py -i /share/magalab/Kat/DSS/qiime/dss.contents_split_library_output/seqs.fna -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set/97_otus.fasta -m usearch61 --usearch61_minh 0.28 --usearch61_xn 8.0 -o /share/magalab/Kat/DSS/qiime/dss.contents_chimera_output

echo "Filtering chimeric sequences."

time filter_fasta.py  -f /share/magalab/Kat/DSS/qiime/dss.feces_split_library_output/seqs.fna -o /share/magalab/Kat/DSS/qiime/dss.feces_chimeras_filtered.fna -s /share/magalab/Kat/DSS/qiime/dss.feces_chimera_output/chimeras.txt -n

time filter_fasta.py  -f /share/magalab/Kat/DSS/qiime/dss.contents_split_library_output/seqs.fna -o /share/magalab/Kat/DSS/qiime/dss.contents_chimeras_filtered.fna -s /share/magalab/Kat/DSS/qiime/dss.contents_chimera_output/chimeras.txt -n

echo "Done Filtering."
```

**Picking Open Reference OTUs**

Using ```pick_open_reference_otus.py```

Script:

```
#!/bin/bash
##

#SBATCH -o /share/magalab/Kat/DSS/qiime/test.out
#SBATCH -e /share/magalab/Kat/DSS/qiime/test.err
#SBATCH -n 4
#SBATCH -p animal_sciences
#SBATCH -t 8-00:00:00

echo "Setting up output and error directories, running 4 ntasks, partitioned to animal_sciences computer."

echo "Loading QIIME"

module load qiime/1.9.1

echo "Picking OTUs for feces and content samples using open reference method."

time pick_open_reference_otus.py -i /share/magalab/Kat/DSS/qiime/dss.feces_chimeras_filtered.fna -o /share/magalab/Kat/DSS/qiime/feces.otus -a -O 8 --min_otu_size 3 -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set/97_otus.fasta

time pick_open_reference_otus.py -i /share/magalab/Kat/DSS/qiime/dss.contents_chimeras_filtered.fna -o /share/magalab/Kat/DSS/qiime/contents.otus -a -O 8 --min_otu_size 3 -r /share/magalab/Kat/DSS/qiime/gg_13_8_otus/rep_set/97_otus.fasta

echo "Done picking OTUs."

echo "Summarizing the number of OTUs per sample"

time biom summarize-table -i /share/magalab/Kat/DSS/qiime/feces.otus/otu_table_mc3_w_tax_no_pynast_failures.biom -o /share/magalab/Kat/DSS/qiime/feces.otus/results_summary.txt

time biom summarize-table -i /share/magalab/Kat/DSS/qiime/contents.otus/otu_table_mc3_w_tax_no_pynast_failures.biom -o /share/magalab/Kat/DSS/qiime/contents.otus/results_summary.txt

echo "Done summarizing."
```

### July 6, 2017

**After checking summary tables, determined that only about 150K reads were actually mapped to OTUs. After more searching, found that the issue began with the demultiplex step. As barcodes may have been compromised with prior trimming steps, I created a new potential workflow to demultiplex first, followed by cleaning and filtering reads for quality. See new workflow.**

Report to show missing reads:

```
klfurtado@cabernet:/share/magalab/Kat/DSS/qiime/dss.feces_split_library_output$ cat histograms.txt
Length	Count
192.0	2
202.0	2
212.0	3
222.0	6
232.0	2
242.0	63
252.0	367
262.0	338
272.0	1104
282.0	6906
292.0	86039
302.0	18
--
```

```
klfurtado@cabernet:/share/magalab/Kat/DSS/qiime/dss.contents_split_library_output$ cat histograms.txt
Length	Count
207.0	2
217.0	3
227.0	7
237.0	6
247.0	163
257.0	206
267.0	523
277.0	1082
287.0	15416
297.0	32157
307.0	0
--
```

Used ```grep -c``` to count reads at various steps to determine where reads were lost. Most lost after using ```split_libraries_fastq.py```
